lib/puppetserver/ca/action/list.rb

```diff
@@ -87,14 +87,15 @@ def run(input)
             filter_names = lambda { |x| true }
           end
 
-          all_certs = get_all_certs(puppet.settings).select { |cert| filter_names.call(cert) }
-          requested, signed, revoked = separate_certs(all_certs)
-          missing = certnames - all_certs.map { |cert| cert['name'] }
-
           if (all || certnames.any?)
+            all_certs = get_all_certs(puppet.settings).select { |cert| filter_names.call(cert) }
+            requested, signed, revoked = separate_certs(all_certs)
+            missing = certnames - all_certs.map { |cert| cert['name'] }
             output_certs_by_state(all, output_format, requested, signed, revoked, missing)
           else
-            output_certs_by_state(all, output_format, requested)
+            all_csrs = get_all_certs(puppet.settings, "requested")
+            missing = []
```

Comment:
is this extra

Reply:
Sort of, it's so that the `return missing.any?` doesn't return a 1 since we're not asking for a specific cert.